app/services/invoices/retry_service.rb

```diff
@@ -78,10 +78,13 @@ def credit_notes
     end
 
     def should_create_credit_note_credit?
+      return false if invoice.one_off?
+
       credit_notes.any?
     end
 
     def should_create_applied_prepaid_credit?
+      return false if invoice.one_off?
```

Comment:
I had a quick and I don't see this condition on the first payment. It seems that you can pay the first time with credit but not when retrying?

Reply:
Actually credit notes / prepaid credits are not applied at all on one off invoices. We don't use those credits in main service so those should be skipped also when retrying